apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/source/StandardGrpcRdaSource.java

```diff
@@ -0,0 +1,237 @@
+              processResult.setException(ex);
+            }
+          } catch (ProcessingException ex) {
+            flushBatch = false;
+            processResult.addCount(ex.getProcessedCount());
+            processResult.setException(ex);
+          } catch (Exception ex) {
+            flushBatch = false;
+            processResult.setException(ex);
+          }
+
+          MultiCloser closer = new MultiCloser();
+
+          if (batch.size() > 0 && flushBatch) {
+            closer.close(() -> processResult.addCount(submitBatchToSink(apiVersion, sink, batch)));
+          }
+
+          closer.close(() -> sink.shutdown(Duration.ofMinutes(5)));
+
+          try {
+            closer.finish();
+          } catch (Exception ex) {
+            if (processResult.getException() != null) {
+              processResult.getException().addSuppressed(ex);
+            } else {
+              processResult.setException(ex);
+            }
+          }
+
+          processResult.addCount(sink.getProcessedCount());
+
+          return processResult;
+        });
+  }
+
+  /**
+   * The RDA API server drops open connections abruptly when it has no data to transmit for some
+   * period of time. These closures are not clean at the protocol level, so they appear as errors to
+   * gRPC, but we don't want to trigger alerts when they happen since they are not unexpected.
+   *
+   * <p>This method determines if we have been idle long enough that such a drop is possible. If the
+   * drop is expected it simply logs the event and returns false to indicate the exception should
+   * not be treated as an error. But if the drop is not expected it returns true to indicate that
+   * the exception should be treated as an error and have normal error logic applied to it.
+   *
+   * @param lastProcessedTime time in millis when we last processed a message from the server
+   * @param exception the exception to evaluate
+   * @return true if not an expected drop and the exception should be treated as an error
+   */
+  private boolean isUnexpectedDroppedConnectionException(
+      long lastProcessedTime, DroppedConnectionException exception) {
+    Exception e = null;
+    final long idleMillis = clock.millis() - lastProcessedTime;
+
+    if (idleMillis >= minIdleMillisBeforeConnectionDrop) {
+      log.info(
+          "RDA API server dropped connection after idle time: idleMillis={} message='{}'",
+          idleMillis,
+          exception.getMessage());
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  /**
+   * Uses the hard coded starting sequence number if one has been configured. Otherwise asks the
+   * sink to provide its maximum known sequence number as our starting point. When all else fails we
+   * start at the beginning. The RDA API assumes we are sending the highest sequence number that we
+   * already have so they will start at that plus 1. For a configured starting sequence number we
```

Comment:
I wonder if long term this `+1` logic will trip someone up like it did us a couple of months ago.  I wonder if it would be better to remove the logic out of this method and put a comment in the configuration yml similar to what you have above.  That way, both the database and the configuration code are designed to have a value 1 below the starting point.  Thoughts?

Reply:
(I know this code was just move from GrpcRdaSource, but just rethinking this now)